### PR TITLE
performed shellcheck for all diagnostics scripts

### PR DESCRIPTION
**Pull request by BenjaminsM**

### CODE REVIEW

File: protocols/Create_Callrate_file_PerSentrixBarcode.sh [hunk 1]
```diff
@@ -82,9 +84,9 @@ perl -pi -e 's|U|Unknown|g' "${tmpCallrateDir}/Callrates_${SentrixBarcode_A}.txt
 
 #Replace barcode with sampleid
 
-barcodelist=()
+#barcodelist=()
```
Comment (pneerincx) [inline protocols/Create_Callrate_file_PerSentrixBarcode.sh:87]: Can be removed?

File: protocols/Create_Callrate_file_PerSentrixBarcode.sh [hunk 2]
```diff
@@ -82,9 +84,9 @@ perl -pi -e 's|U|Unknown|g' "${tmpCallrateDir}/Callrates_${SentrixBarcode_A}.txt
...
 
-n_elements=${Sample_ID[@]}
+#n_elements=${Sample_ID[@]}
```
Comment (pneerincx) [inline protocols/Create_Callrate_file_PerSentrixBarcode.sh:89]: Can be removed?

File: scripts/Make_Callrate_Report.py [hunk 3]
```diff
@@ -1,3 +1,5 @@
+#!/bin/python
```
Comment (pneerincx) [inline scripts/Make_Callrate_Report.py:1]:
```suggestion
#!/usr/bin/env python
```

Comment (pneerincx) [reply scripts/Make_Callrate_Report.py:1]: It is relatively safe to use a hardcoded path to the Bash interpreter as we have that on all machines. Python is a different story tough and there are major differences between Python 2 and 3. If you use
`module load Python/some-specific-version`
and then a script that starts with
`#!/usr/bin/env python`
, the script will run with the Python version loaded by the module system and fetched from the environment.
If you instead use the hard coded path /bin/python, then the script will run with the Python version installed by the Linux distro and ignore whatever was `module load`ed.

File: protocols/CalculateSD.sh [hunk 4]
```diff
@@ -48,7 +50,7 @@ sd=$(echo "${log_ratios[@]}" | awk '{sum+=$1; sumsq+=$1*$1}END{print sqrt(sumsq/
 echo "${sd}" > "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.FINAL.vcf.sd"
 
 mv "${tmpCalculateSDDir}/${SentrixBarcode_A}_${SentrixPosition_A}.vcf" "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.vcf"
-awk '{OFS="\t"}{if ($0 ~ "#CHROM" ){ print $1,$2,$3,$4,$5,$6,$7,$8,$9,"'${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}'"} else {print $0}}' "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.vcf" > "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.FINAL.vcf"
+awk '{OFS="\t"}{if ($0 ~ "#CHROM" ){ print $1,$2,$3,$4,$5,$6,$7,$8,$9,"'"${Sample_ID}"_"${SentrixBarcode_A}"_"${SentrixPosition_A}"'"} else {print $0}}' "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.vcf" > "${tmpCalculateSDDir}/${Sample_ID}_${SentrixBarcode_A}_${SentrixPosition_A}.FINAL.vcf"
```
Comment (pneerincx) [inline protocols/CalculateSD.sh:53]: That is a lot of single and double commas; was this tested?

**Review by pneerincx — CHANGES_REQUESTED** [review]
See inline comments.

File: protocols/CalculateSD.sh (hunk 4 above)
Comment (BenjaminsM) [reply protocols/CalculateSD.sh:53]: we should test this indeed, we still need to do this . however i need to make some other changes also to the gap pipeline so il will make those too and test them all at once :-)

**Review by pneerincx — APPROVED** [review]

**Pull request merged**